source/extensions/http/header_formatters/preserve_case/preserve_case_formatter.cc

```diff
@@ -12,20 +12,34 @@ namespace Http {
 namespace HeaderFormatters {
 namespace PreserveCase {
 
-PreserveCaseHeaderFormatter::PreserveCaseHeaderFormatter(const bool forward_reason_phrase)
-    : forward_reason_phrase_(forward_reason_phrase) {}
+PreserveCaseHeaderFormatter::PreserveCaseHeaderFormatter(
+    const bool forward_reason_phrase,
+    const envoy::extensions::http::header_formatters::preserve_case::v3::
+        PreserveCaseFormatterConfig::FormatterTypeOnUnknownHeaders
+            formatter_type_on_unknown_headers)
+    : forward_reason_phrase_(forward_reason_phrase),
+      formatter_type_on_unknown_headers_(formatter_type_on_unknown_headers) {
+  switch (formatter_type_on_unknown_headers_) {
+  case envoy::extensions::http::header_formatters::preserve_case::v3::PreserveCaseFormatterConfig::
+      PROPER_CASE:
+    header_key_formatter_on_unknown_headers_ =
+        std::make_unique<Envoy::Http::Http1::ProperCaseHeaderKeyFormatter>();
+    break;
+  default:
```

Comment:
It would be better to handle every enum case and let the default to become a fall through or exception catcher.

Reply:
done

Reply:
Ping on this. Default I think in this case should throw an exception since it happens on config load. Please do that and add a test for that case.

Reply:
done